src/plugins/intel_cpu/src/nodes/dft.cpp

```diff
@@ -289,139 +314,279 @@ void DFT::dftNd(float* output, const std::vector<size_t>& outputStrides) const {
+                sumImag *= reciprocalNComplex;
+            }
+            outputBuffer[k * 2] = sumReal;
+            outputBuffer[k * 2 + 1] = sumImag;
+        };
+    }
 
-        if (inverse) {
-            sumReal /= nComplex;
-            sumImag /= nComplex;
-        }
-        outputBuffer[k * 2] = sumReal;
-        outputBuffer[k * 2 + 1] = sumImag;
-    });
+    parallel_for(nComplex, blockIteration);
     cpu_memcpy(data, outputBuffer.data(), dataLength * sizeof(float));
 }
 
-std::vector<std::pair<float, float>> DFT::generateTwiddles(size_t n_complex) const {
-    std::vector<std::pair<float, float>> twiddles(n_complex * n_complex);
+std::vector<float> DFT::generateTwiddlesDFT(size_t n_complex, bool inverse) const {
+    std::vector<float> twiddles(n_complex * n_complex * 2);
+    const float inverseMultiplier = inverse ? 1 : -1;
     parallel_for(n_complex, [&](const size_t k) {
         for (size_t n = 0; n < n_complex; ++n) {
-            float phase =  2.0f * PI * static_cast<float>(n * k) / static_cast<float>(n_complex);
+            float phase = 2.0f * PI * static_cast<float>(n * k) / static_cast<float>(n_complex);
             auto complexReal = std::cos(phase);
-            auto complexImag = -std::sin(phase);
-            twiddles[k * n_complex + n] = std::make_pair(complexReal, complexImag);
+            auto complexImag = std::sin(phase) * inverseMultiplier;
+            twiddles[2 * (k * n_complex + n)] = complexReal;
+            twiddles[2 * (k * n_complex + n) + 1] = complexImag;
         }
     });
     return twiddles;
 }
 
+void DFT::updateTwiddlesFFT(size_t n_complex, bool inverse) {
+    const float inverseMultiplier = inverse ? 1 : -1;
+    size_t numBlocks = 1;
+
+    twiddlesFFT.reserve((n_complex - 1) * 2);
+    if (twiddlesFFT.size() == 0) {
+        twiddlesFFT.emplace_back(1.0f);   //  cos(0)
+        twiddlesFFT.emplace_back(-0.0f);  // -sin(0)
+    } else {
+        for (size_t i = numBlocks; i < twiddlesFFT.size() / 2; i += numBlocks) {
+            numBlocks *= 2;
+        }
+    }
+
+    for (size_t i = twiddlesFFT.size() / 2; i < n_complex - 1; i += numBlocks) {
+        numBlocks *= 2;
+
+        for (size_t blockNum = 0; blockNum < numBlocks; blockNum++) {
+            size_t copyIndex = twiddlesFFT.size() - blockNum - numBlocks;
+
+            twiddlesFFT.push_back(twiddlesFFT[copyIndex]);
+            twiddlesFFT.push_back(twiddlesFFT[copyIndex + 1]);
+
+            blockNum++;
+
+            float angle = PI * blockNum / numBlocks;
+            auto complexReal = std::cos(angle);
+            auto complexImag = std::sin(angle) * inverseMultiplier;
+
+            twiddlesFFT.emplace_back(complexReal);
+            twiddlesFFT.emplace_back(complexImag);
+        }
+    }
+}
+
 bool DFT::created() const {
     return getType() == Type::DFT;
 }
 
-void DFT::createPrimitive() {}
+void DFT::prepareParams() {
+    bool hasDFT = false;
+    bool hasFFT = false;
```

Comment:
Using two variables is redundant, we usually use only one algo easier FFT or DFT, so one flag (avaliableFFT) should be enough, since when FFT is not applicable, DFT is the only option.

Reply:
No, FFT is not used for non-power-of-two dimensions, so, even when hasFFT=true we may still need DFT.

Reply:
DFT operation in OpenVINO can handle few axes at once - and these axes may have different shapes (e.g., 2 axes, 256 and 200).
FFT is selected for power-of-2 shapes only, DFT is used for others.
For the case in example we will need both kernels - FFT for the first (256) and DFT for the second (200).

Reply:
The algo selection is done inside DFT::dftNd() function.

Reply:
Ok. Got it, thank you!